### PR TITLE
dev/core#4754 - Add status check to warn of missing/outdated public assets

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This warns admins about assets that should have been copied by Composer but weren't.

See https://lab.civicrm.org/dev/core/-/issues/4754

Comments
-----------
Need some docs to link to, otherwise the warning message isn't extremely helpful.

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
@colemanw I gave this an R-run on my local d9 and it works for me once you have also updated your civicrm-asset-plugin version I have also opened https://lab.civicrm.org/documentation/docs/sysadmin/-/merge_requests/370 to give some clarification on this matter which could be used as a link in this status message

**Comment by colemanw** [comment]
Thanks @seamuslee001 I've merged your docs and added the link to this PR.

File: CRM/Utils/Check/Component/Cms.php [hunk 1]
```diff
@@ -16,6 +16,45 @@
...
+      $assetVersion = json_decode(file_get_contents($versionFile), TRUE);
+      if ($assetVersion !== $currentVersion) {
+        $errorMsg = ts('Wrong version of assets in public web directory: expected %1 but found %2.', [1 => $currentVersion, 2 => $assetVersion]);
+      }
+    }
+    if ($errorMsg) {
+      $checks[] = new CRM_Utils_Check_Message(
+        __FUNCTION__,
+        $errorMsg .
+        '<br /><a target="_blank" href="https://docs.civicrm.org/sysadmin/en/latest/upgrade/drupal8/#ensure-that-civicrm-assets-are-up-to-date">' .
```
Comment (demeritcowboy) [inline CRM/Utils/Check/Component/Cms.php:47]: Docs link has already gone stale - it's now `#additional-cleanup`.
Do you want to update that and then I'd merge this.

Comment (demeritcowboy) [reply CRM/Utils/Check/Component/Cms.php:47]: Also should probably use docURL2, like at https://github.com/civicrm/civicrm-core/blob/5c4ecf855000ecbdfc82782f01c3cdadf76212c4/CRM/Utils/Check/Component/Env.php#L123

Comment (colemanw) [reply CRM/Utils/Check/Component/Cms.php]: @demeritcowboy updated, thanks for the suggestion

File: CRM/Utils/Check/Component/Cms.php [hunk 2]
```diff
@@ -16,6 +16,43 @@
...
+      $currentVersion = CRM_Utils_System::version();
+      $assetVersion = json_decode(file_get_contents($versionFile), TRUE);
+      if ($assetVersion !== $currentVersion) {
+        $errorMsg = ts('Wrong version of assets in public web directory: expected %1 but found %2.', [1 => $currentVersion, 2 => $assetVersion]);
+      }
+    }
+    if ($errorMsg) {
+      $checks[] = new CRM_Utils_Check_Message(
+        __FUNCTION__,
+        "$errorMsg Use the command <code>composer civicrm:publish</code> to resolve. " .
```
Comment (demeritcowboy) [inline CRM/Utils/Check/Component/Cms.php:46]: I think a `ts` went missing. (Nobody _should_ be seeing this message to begin with but my magic wand is in the shop.)

Comment (demeritcowboy) [reply CRM/Utils/Check/Component/Cms.php]: D'oh now there's no space between the sentence and the Learn More link.

Comment (colemanw) [reply CRM/Utils/Check/Component/Cms.php]: ts(' ')

Comment (colemanw) [reply CRM/Utils/Check/Component/Cms.php]: ts("😆")

**Comment by demeritcowboy** [comment]
I get a syntax error. Unless that's a feature in a newer version of php 8.x, where you can ping someone every time the line of code runs. That would be a terrible feature.

**Pull request merged**